src/day8/re_frame/trace/app_state.cljs

```diff
@@ -0,0 +1,54 @@
+(ns day8.re-frame.trace.app-state
+  (:require [reagent.core :as r]
+            [clojure.string :as str]
+            [devtools.formatters.core :as cljs-devtools]
+
+            [cljs.pprint :refer [pprint]]))
+
+(defn string->css [css-string]
+  (->> (str/split (get css-string "style") #";")
+       (map #(str/split % #":"))
+       (reduce (fn [acc [property value]]
+                 (assoc acc (keyword property) value)) {})))
+
+(defn str->hiccup
+  [string]
+  (cond (= string "span")   :span
+        (= string "style")  :style
+        (= string ", ")     " "
+        :else               string))
+
+(declare jsonml->hiccup)
+
+(defn data-structure
+  [jsonml]
+  (let [expanded? (r/atom true)]
```

Comment:
I think you need to also use [`has-body-api-call`](https://github.com/binaryage/cljs-devtools/blob/b42ad914da02cd71d9765d32f54007cb9e2ca68f/src/lib/devtools/formatters/core.cljs#L84.) to check whether something is expandable? If it's not, then don't show the toggles?